### DISCUSSION TITLE
Create an endpoint for fetching roles

### DISCUSSION THREAD

**Discussion by rishighan** [bpepple/metron#83] [API Suggestions]
Hello, currently as a part of integrating with Metron API through my app, I am populating the dropdown for fetching and auto-suggesting the `creators`, and was wondering if there is a similar endpoint for `roles`.

Note that in all dropdowns, the user can create/add their own values.

Illustration: 

![Screen Shot 2022-01-05 at 8 10 47 PM](https://user-images.githubusercontent.com/414927/148326905-880763a8-7ff7-43dc-9d53-5c55d7146dd3.png)

Couple of points to think about:

1. I think there is value in user adding their own data for fields such as `creators` and `roles` and Metron validating it to some degree. 
2. It follows from point 1 that we have some validations on the UI end too, such as character limits, etc.

What are your thoughts @bpepple ?

**Comment by bpepple** [comment]
Based on your user-case, I think it does makes sense to add an endpoint to retrieve creator roles from Metron. Also, to implement it there is almost zero impact on existing tools since we are just adding a new resource and not modifying an existing one.

I'm going to create an issue for this, since the impact and effort to implement is pretty minimal.

**Comment by bpepple** [comment]
Went ahead and made a pull request for this in #85, and if you get a chance please give it a glance to see if there are any changes you might like to suggest. Otherwise I will probably merge it and update the website tomorrow. Thx.

**Reply by rishighan** [reply]
Just checked it out, looks good!

**Reply by bpepple** [reply]
Cool. There's not really much to it, but wanted to make sure it works for you before pushing to production.

**Reply by rishighan** [reply]
Much appreciated :) On my end, it involves [removing the check for `__str__` in the response](https://github.com/rishighan/threetwo/blob/0bfe5803892d651f1a887d6c3cc69ae9d22dd54d/src/client/actions/metron.actions.tsx#L14)

**Reply by bpepple** [reply]
Ok, this has been pushed to production.

**Reply by rishighan** [reply]
Just tested it and works perfectly.

![Screen Shot 2022-01-06 at 9 00 43 AM](https://user-images.githubusercontent.com/414927/148421101-d0ddff84-278b-4188-ad49-eaed4e94dda2.png)
